packages/bot-web-ui/src/components/dashboard/bot-builder/toolbox/search-box/search-icon.tsx

```diff
@@ -0,0 +1,16 @@
+import React from 'react';
+import { Icon } from '@deriv/components';
+
+type TSearchIcon = {
+    search: string;
+    is_search_loading: boolean;
+    onClick: () => void;
+};
+
+const SearchIcon = ({ search, is_search_loading, onClick }: TSearchIcon) => {
+    if (!search) return <Icon icon='IcSearch' />;
+    if (is_search_loading) return <div className='loader' />;
```

Comment:
There is a question:
Can `is_search_loading` be true also in the same time `search` be false?
If yes you still need to check if `search` is true.

Reply:
reverted the changes

Reply:
@Bahar-FS 

the `search` parameter is the actual value (text/string) of search. 
`is_search_loading` cannot be true if there is no search value.

so if there is no search value show the search icon.
If there is search value and is searching show the progress icon
if there is search value and is not searching show the clear icon


https://user-images.githubusercontent.com/90243468/195775309-66f0a84b-f1a5-4815-860c-f8f14350fa29.mov